trucks.js

```diff
@@ -503,9 +505,80 @@ var foodTrucks = [
   }
 ];
 
-// this module should support the following methods:
-// getTrucks() - return all trucks
+// this module should support the following methods: getTrucks() - return all trucks
+var getTrucks = function() {
+  var allTrucks = [];
+  for (var i = 0; i < foodTrucks.length; i++) {
+    allTrucks.push(foodTrucks[i].name );
+  }
+
+  return allTrucks;
+};
+
 // getTruck(name) - return the truck object matching 'name'
+var getTruck = function(name) {
+  var thisTruck =[];
+  for (var i = 0; i < foodTrucks.length; i++) {
+    var nameOfTruck = foodTrucks[i].name.toString();
+    if (name === nameOfTruck) {
+      thisTruck.push( foodTrucks[i]);
```

Comment:
Instead of pushing the found truck into an array, just return the truck directly. The caller is expecting a truck object back, not an array.